src/utils/server-utils.ts

```diff
@@ -52,6 +52,46 @@ function formatCapabilitiesInfo(
   }, {});
 }
 
+type FlatLayer = {
+  Name: {
+    _text: string;
+    [key: string]: any;
+  };
+  Dimension: {
+    _text: string;
+    [key: string]: any;
+  };
+  [key: string]: any;
+};
+
+type FlatLayerContainer = { Layer: FlatLayer; [key: string]: any };
+type LayerContainer =
+  | { Layer: LayerContainer; [key: string]: any }
+  | FlatLayerContainer[]
+  | FlatLayer[];
+
+const isArrayOfFlatLayerContainers = (
+  maybeArray: LayerContainer,
+): maybeArray is FlatLayerContainer[] => {
+  return (maybeArray as FlatLayerContainer[])[0].Layer !== undefined;
+};
+
+function flattenLayers(rawLayers: LayerContainer): FlatLayer[] {
+  if ('Layer' in rawLayers) {
```

Comment:
Generally, we don't like using `in` to check for properties in Javascript, because it actually checks for properties that exist anywhere on this object's prototype chain, not just this object itself. It may well not matter in this particular case, but it's not considered good practice.

The "right" way to do this is to use `hasOwnProperty` (could either be: `rawLayers.hasOwnProperty('Layer')` or event "better": `Object.prototype.hasOwnProperty.call(rawLayers, 'Layer')` (the latter is preferred because if an object is created in a certain way, e.g. `const myObj = Object.create(null)`, `hasOwnProperty` might not be defined on it - again probably doesn't matter in this case, but it's the "right" way to do it).

You can also use `lodash#has` for this, which takes care of doing this check the "right" way in an easier-to-remember function call. JS is awesome ;-).